repos/system_upgrade/el7toel8/actors/cupscheck/actor.py

```diff
@@ -0,0 +1,226 @@
+      """
+      has_directive
+
+      Checks if configuration file has directive set.
+
+      parameters:
+      - directives_list - list of strings
+      - filename - string
+
+      returns boolean
+      """
+      found = False
+
+      with open(filename, 'r') as opened_config:
+        file_contents = opened_config.read()
+        for directive in directives_list:
+          # Digest/BasicDigest are directive values, so regular expression differs
+          if directive == 'Digest' or directive == 'BasicDigest':
+            regexp_string = '\s+'
+          else:
+            regexp_string = '^\s*'
+          regexp_string = regexp_string + directive + '\s'
+          pattern = compile(regexp_string, MULTILINE)
+          matched_list = findall(pattern, file_contents)
+          if len(matched_list) > 0:
+            found = True
+            break
+        opened_config.close()
+
+      return found
+
+    def look_for_affected_features(self, include_files_list=[CUPSD_CONF]):
+      """
+      look_for_affected_features
+
+      parameters:
+      - include_file_list - list of strings
+
+      returns dictionary
+      """
+
+      matrix = {
+                  'interface' : False,
+                  'digest' : False,
+                  'include' : False,
+                  'certkey' : False,
+                  'env' : False,
+                  'printcap' : False
+                }
+
+      matrix['interface'] = self.has_interface_script()
+
+      for change_list in cupsd_changes_list:
+        for config_file in include_files_list:
+          # If we already found the directive somewhere - in regular configuration file or
+          # included one - we do not need to search anymore, the migration will happen.
+
+          # Names like KEY, DIRECTIVES, SOURCE are just enum constants corresponding with
+          # scheme in libraries/py - just for better understanding what lies on
+          # specific indexes of the list (hope)
+          if matrix[change_list[KEY]] is False:
+            matrix[change_list[KEY]] = self.has_directive(change_list[DIRECTIVES], config_file)
+
+      for change_list in cupsfiles_changes_list:
+        matrix[change_list[KEY]] = self.has_directive(change_list[DIRECTIVES], change_list[SOURCE])
+
+      return matrix
+
+    def get_include_files_list(self):
+      """
+      get_include_files_list
+
+      we need all included files in list
+
+      returns list of strings
+      """
+      paths = [CUPSD_CONF]
+      pattern = compile('^\s*Include\s+(.+?)\s', MULTILINE)
+
+      for found_path in paths:
+        with open(found_path, 'r') as opened_config:
+          file_contents = opened_config.read()
+          matched_list = findall(pattern, file_contents)
+          if len(matched_list) > 0:
+            for new_path in matched_list:
+              if new_path not in paths:
+                paths.append(new_path)
+          opened_config.close()
+
+      return paths
+
+    def process(self):
```

Comment:
No need for `is True` or `is False` at if statements.

Reply:
Corrected.